### DISCUSSION TITLE
Why the lackluster example mod?

### DISCUSSION THREAD

**Discussion by Darthsae** [neoforged/NeoForge#631] [General]
I've always been confused about why Minecraft modding platforms' example mods are so, lackluster, and unhelpful, while tmodloaders' is practically a full-fledged documentation of what you can do, is there a reason for this?

**Comment by PaintNinja** [comment]
Maintainability - I requested this long before the split and at the time was told it would be too much work to maintain and that people need to figure out things for themselves to a degree instead of becoming overly reliant on MC-specific example code.

Both Forge and NeoForge's MDK contain examples for core features everyone's expected to know about however - namely registration, configs and event handling.

**Reply by Darthsae** [reply]
I guess so, but the documentation is kind of hard to use as well, it would really need to be better for that to work, it doesn't even have a list of all methods and variables, so if you want to find that you have to dig through the source code. They could at least show the example block and item if you make a class, and there isn't an example entity, which feels kind of important.

**Reply by KnightMiner** [reply]
You have to dig through the source code to do most of anything with modding. Forge is not a complete API that handles all your modding needs, its a compatibility layer on top of undocumented Minecraft source code. If you cannot work with Minecraft source code, you will have a hard time doing most things in modding.

As far as example blocks and items go, thats just basic object oriented stuff, if you understand registration (which is in the example mod) then you have everything you need to make blocks and items.

Example entities are complex enough that you will not find a one size fits all example, projectiles are wildly different from mobs which are wildly different from things like item frames. Just use vanilla as an example, vanilla is a great example.

**Reply by Darthsae** [reply]
I understand that, but that could at least have an example, any starting place is better than literally nothing.

**Reply by KnightMiner** [reply]
Vanilla has hundreds of examples, I use them on a regular basis
